### PR TITLE
feat(api): add support to pass custom request header

### LINKED ISSUE #17: Single method for e-wallet payment creation
Currently, Xendit API supports OVO, DANA, and LINKAJA. The current method implementation is separated into 3 different methods: `createOvoPayment`, `createLinkajaPayment`, and `createOvoPayment`. Such implementations chosen because of the API responses for create e-wallet payment are not standardized yet.

For further improvement, it might be better if the implementation is just in one single method, like `createPayment` as it will possibly expands to supports more e-wallets in the future.

### LINKED ISSUE #43: X-IDEMPOTENCY-KEY  implementation required
Required implementation for `X-IDEMPOTENCY-KEY` to prevent processing duplicate requests.

### LINKED ISSUE #44: Support headers in network package
Current implementation of network client doesn't support headers, causing problems like #43 
https://github.com/xendit/xendit-java/blob/master/xendit-java-lib/src/main/java/com/xendit/network/BaseRequest.java

### LINKED ISSUE #52: Support for-user-id (xenPlatform enabled account)
Hi, can we add support to enable custom header?

E.g. passing **for-user-id** in request header to create invoice for sub-account as mentioned in https://xendit.github.io/apireference/#create-invoice

### PR DESCRIPTION
**Pull request by kahirul**
1. Add support to pass custom request header. Resolves #52, Resolves #44, Resolves #43
2. Add `createEWalletPayment`. Resolves #17

### CODE REVIEW

**Comment by stanleynguyen** [comment]
@kahirul Thank you so much for your contribution

File: xendit-java-lib/src/main/java/com/xendit/model/EWalletPayment.java [hunk 1]
```diff
@@ -136,7 +136,20 @@ public static EWalletPayment createDanaPayment(
...
+
+  /**
+   * Create new ewallet payment with all parameters as HashMap
+   *
+   * @param headers
+   * @param params mount end-customer will pay.
+   * @return EWalletPayment
+   * @throws XenditException XenditException
+   */
+  public static EWalletPayment createEWalletPayment(
```
Comment (stanleynguyen) [inline xendit-java-lib/src/main/java/com/xendit/model/EWalletPayment.java:150]: Can this be renamed as `createPaymentRequest`?

Comment (kevindavee) [reply xendit-java-lib/src/main/java/com/xendit/model/EWalletPayment.java:150]: How about `createPayment` ? Payment request sounds like someone needs to approve the payment request from the other side 😁

Comment (stanleynguyen) [reply xendit-java-lib/src/main/java/com/xendit/model/EWalletPayment.java:150]: No it's meant to be an override of the existing method. I think consistency is more important

Comment (stanleynguyen) [reply xendit-java-lib/src/main/java/com/xendit/model/EWalletPayment.java:150]: Also paymentRequest is the right term

**Comment by kahirul** [comment]
> @kevindavee Please help release as 1.7.0 after requested change has been made

I'm not familiar with release process. Do you have docs about this somewhere?

**Comment by kahirul** [comment]
I know this is not a support page.

But just curious how long will it take for this changes to land in Xendit bintray repository?
@kevindavee

**Comment by kevindavee** [comment]
Hi @kahirul . 

It's not that long. But I'm away from my computer right now. Do you mind to wait at worse 4.30 PM Jakarta Time? I'll get to my computer as soon as possible

**Comment by kahirul** [comment]
@kevindavee awesome. That's fine for me. Thank you

**Comment by kevindavee** [comment]
@kahirul `xendit-java` version `1.7.0` has been published. Thanks for contributing to this library!

**Pull request merged**